ukis_pysat/data.py

```diff
@@ -30,6 +35,144 @@
 from ukis_pysat.file import env_get
 from ukis_pysat.members import Datahub
 
+BASE_URL = (
+    "https://storage.googleapis.com/gcp-public-data-landsat/"
+    "{sensor}/{collection:02}/{path:03}/{row:03}/{product_id}/"
+)
+
+
+def meta_from_pid(product_id):
+    """Extract metadata contained in a Landsat Product Identifier."""
+    meta = {}
+    parts = product_id.split("_")
+    meta["product_id"] = product_id
+    meta["sensor"], meta["correction"] = parts[0], parts[1]
+    meta["path"], meta["row"] = int(parts[2][:3]), int(parts[2][3:])
+    meta["acquisition_date"] = dt.strptime(parts[3], "%Y%m%d")
+    meta["processing_date"] = dt.strptime(parts[4], "%Y%m%d")
+    meta["collection"], meta["tier"] = int(parts[5]), parts[6]
+    return meta
+
+
+def compute_md5(fpath):
+    """Get hexadecimal MD5 hash of a file."""
+    with open(fpath, "rb") as f:
+        h = hashlib.md5(f.read())
+    return h.hexdigest()
+
+
+def download_files(url, outdir, progressbar=False, verify=False):
+    """Download a file from an URL into a given directory.
+
+    Parameters
+    ----------
+    url : str
+        File to download.
+    outdir : str
+        Path to output directory.
+    progressbar : bool, optional
+        Display a progress bar.
+    verify : bool, optional
+        Check that remote and local MD5 haches are equal.
+
+    Returns
+    -------
+    fpath : str
+        Path to downloaded file.
+    """
+    fname = url.split("/")[-1]
+    fpath = os.path.join(outdir, fname)
+    r = requests.get(url, stream=True)
+    remotesize = int(r.headers.get("Content-Length", 0))
+    etag = r.headers.get("ETag", "").replace('"', "")
+
+    if r.status_code != 200:
+        raise requests.exceptions.HTTPError(str(r.status_code))
+
+    if os.path.isfile(fpath) and os.path.getsize(fpath) == remotesize:
+        return fpath
+    if progressbar:
+        progress = tqdm(total=remotesize, unit="B", unit_scale=True)
+        progress.set_description(fname)
+    with open(fpath, "wb") as f:
+        for chunk in r.iter_content(chunk_size=1024 * 1024):
+            if chunk:
+                f.write(chunk)
+                if progressbar:
+                    progress.update(1024 * 1024)
+
+    r.close()
+    if progressbar:
+        progress.close()
+
+    if verify:
+        if not compute_md5(fpath) == etag:
+            raise requests.exceptions.HTTPError("Download corrupted.")
+
+    return fpath
+
+
+class Product:

```

Comment:
Please be more descriptive here as well, it is only for Landsat. 
And add you sources!